Sources/NIOUDPEchoClient/main.swift

```diff
@@ -0,0 +1,147 @@
+ //===----------------------------------------------------------------------===//
+ //
+ // This source file is part of the SwiftNIO open source project
+ //
+ // Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+ // Licensed under Apache License v2.0
+ //
+ // See LICENSE.txt for license information
+ // See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+ //
+ // SPDX-License-Identifier: Apache-2.0
+ //
+ //===----------------------------------------------------------------------===//
+ import NIO
+ 
+ print("Please enter line to send to the server")
+ let line = readLine(strippingNewline: true)!
+ 
+ private final class EchoHandler: ChannelInboundHandler {
+    public typealias InboundIn = AddressedEnvelope<ByteBuffer>
+    public typealias OutboundOut = AddressedEnvelope<ByteBuffer>
+    private var numBytes = 0
+    
+    private let remoteAddressInitializer: () throws -> SocketAddress
+    
+    init(remoteAddressInitializer: @escaping () throws -> SocketAddress) {
+        self.remoteAddressInitializer = remoteAddressInitializer
+    }
+
+    public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let envelope = self.unwrapInboundIn(data)
+        var byteBuffer = envelope.data
+
+        numBytes -= byteBuffer.readableBytes
```

Comment:
we use explicit `self.` so if you wouldn't mind changing this to `self.numBytes -= byteBuffer.readableBytes`

Reply:
Sure, this is exactly the same as the TCP client. I think it is a good idea to keep them as similar as possible, so I will raise another PR to make the change there too.